packages/web/src/components/TeeResults.vue

```diff
@@ -153,6 +170,46 @@ const updateDetailResult = (id: string | number) => {
   programs.setDetailResult(id, props.trackId)
 }
 
+interface costInfos {
+  prefix: string
+  text: string | undefined
+}
+const getCostInfos = (program: ProgramData) => {
+  let obj: costInfos = {
+    prefix: '',
+    text: ''
+  }
+  // console.log('TeeResults > onBeforeMount > resultsProgs :', resultsProgs )
+  switch (program["nature de l'aide"]) {
+    case 'accompagnement':
+      obj.prefix = 'programCosts.costPrefix'
+      obj.text = program["coût de l'accompagnement"]
+      break
+    case 'formation':
+      obj.prefix = 'programCosts.costPrefix'
+      obj.text = program["coût de l'accompagnement"]
+      break
+    case 'financement':
+      obj.prefix = 'programCosts.aidPrefix'
+      obj.text = program['montant du financement']
+      break
+    case 'prêt':
+      obj.prefix = 'programCosts.loan'
+      obj.text = program['montant du prêt']
+      break
+  }
+  return obj
+}
+
+const getCostInfosPrefix = (program: ProgramData) => {
+  const infos = getCostInfos(program)
```

Comment:
thought: I'm thinking about performance and here the `getCostInfos` is call twice to get the same thing. Maybe it could be better to just call this function from the view and get properties from `costInfos` object result like:
```vue
{{ choices.t(getCostInfosPrefix(prog)) }} :
{{ getCostInfosText(prog) }}
```
replace by
```vue
{{ getCostInfos(prog).prefix) }} :
{{ getCostInfos(prog).text }}
```

What do you think about this ?

Reply:
My advise to find a solution is wrong !!

Reply:
I think this function `getCostInfos` could return the whole string 

...no interface, no middleman, just something like :

```js
const getCostInfos = (program: ProgramData) => {
  let prefix: string = ''
  let text: string | undefined = ''
  // console.log('TeeResults > onBeforeMount > resultsProgs :', resultsProgs )

  switch (program["nature de l'aide"]) {
    case ProgramAidType.acc:
      prefix = 'programCosts.costPrefix'
      text = program["coût de l'accompagnement"]
      break
    case ProgramAidType.train:
      prefix = 'programCosts.costPrefix'
      text = program["coût de l'accompagnement"]
      break
    case ProgramAidType.fund:
      prefix = 'programCosts.aidPrefix'
      text = program['montant du financement']
      break
    case ProgramAidType.loan:
      prefix = 'programCosts.loan'
      text = program['montant du prêt']
      break
  }
  // Translate prefix
  prefix = choices.t(prefix)

  return `${prefix} : ${text}`
}
```

WDYT ?